src/pages/index.js

```diff
@@ -0,0 +1,337 @@
+   .then(()=> {
+    evt.target.classList.toggle("card__like-button_liked"); })
+    .catch(console.error);
+  } ;
+
+function handleDeleteSubmit(evt){
+  evt.preventDefault();
+
+  const deleteModalDeleteBtn = evt.submitter;
+  setButtonText(deleteModalDeleteBtn,true,'Delete','Deleting...');
+
+  api.removeCard(selectedCardId)
+  .then(()=> {
+  selectedCard.remove();
+  closeModal(deleteModal);
+  setButtonText(deleteModalDeleteBtn,false,'Delete','Deleting...');
+  })
+  .catch((err) => {
+    console.error(err);
+    setButtonText(deleteModalDeleteBtn,false,'Delete','Deleting...');
+  });
+};
+
+
+deleteForm.addEventListener("submit", handleDeleteSubmit);
+
+function getCardElement(data) {
+  const cardElement = cardTemplate.content
+    .querySelector(".card")
+    .cloneNode(true);
+  const cardNameElement = cardElement.querySelector(".card__title");
+  const cardImageElement = cardElement.querySelector(".card__image");
+  const cardLikeBtn = cardElement.querySelector(".card__like-button");
+  const cardDeleteBtn = cardElement.querySelector(".card__delete-button");
+
+
+  cardNameElement.textContent = data.name;
+  cardImageElement.src = data.link;
+  cardImageElement.alt = data.name;
+
+  if(data.isLiked){cardLikeBtn.classList.add("card__like-button_liked")};
+
+  cardLikeBtn.addEventListener("click", (evt)=> handleLike(evt, data._id));
+
+  cardDeleteBtn.addEventListener("click", () => handleDeleteCard(cardElement,data._id));
+
+
+
+  cardImageElement.addEventListener("click", () => {
+    previewModalImage.src = cardImageElement.src;
+    previewModalImage.alt = cardImageElement.alt;
+    previewModalCaption.textContent = cardNameElement.textContent;
+    openModal(previewModal);
+  });
+
+  return cardElement;
+}
+
+previewModalCloseBtn.addEventListener("click", () => {
+  closeModal(previewModal);
+});
+
+function openModal(modal) {
+  modal.classList.add("modal_open");
+  document.addEventListener("keydown", handleEscape);
+}
+
+function closeModal(modal) {
+  modal.classList.remove("modal_open");
+  document.removeEventListener("keydown", handleEscape);
+}
+
+profileEditBtn.addEventListener("click", () => {
+  modalNameInput.value = profileTitle.textContent;
+  modalDescriptionInput.value = profileDescription.textContent;
+  openModal(editProfileModal);
+  resetValidation(editModalFormElement, settings);
+});
+
+editModalCloseBtn.addEventListener("click", () => {
+  closeModal(editProfileModal);
+});
+
+function handleModalFormSubmit(evt) {
+  evt.preventDefault();
+
+  const modalSubmitBtn = evt.submitter;
+   setButtonText(modalSubmitBtn,true);
+
+   api.editUserInfo({name:modalNameInput.value,about:modalDescriptionInput.value})
+    .then((data) => {
+  profileTitle.textContent = data.name;
+  profileDescription.textContent = data.about;
+  closeModal(editProfileModal);
+  setButtonText(modalSubmitBtn,false);
+    })
+    .catch((err) => {
+      console.error(err);
+      setButtonText(modalSubmitBtn,false);
```

Comment:
`setButtonText(modalSubmitBtn,false);` 

It should be only in `.finally`, to avoid code duplication in `.then` and `.catch`

**Please, correct it everywhere in the code**